K*(892) optimized

murioni opened the pull request:
-adding a new task for K*(892)  analysis in PbPb using filters and partitions on ResonanceTables 
-adding TrackOccupancyInTimeRange and ImpactParameter to ResonanceTables

BongHwi commented:
Hello @murioni ,
Can we consider adding an option of trackOccupancyInTimeRange cut on the initializer or CollisionCuts?
One of the main idea of using the initializer is to handle all event selections on the initializer and minimize the analysis task.
If the trackOccupancyInTimeRange value is needed to be computed in the analysis task, we can save it on the collision table.

alibuild commented:
Error while checking build/O2Physics/o2 for 11ed463bd5be8f0496ad29fda72615e591878020 at 2024-07-24 20:16:
```
## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:651:63: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:665:70: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:679:66: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:692:67: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:705:91: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:718:98: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:731:99: error: unused parameter 'resotracks' [-Werror=unused-parameter]
/sw/SOURCES/O2Physics/6978-slc7_x86-64/0/PWGLF/Tasks/Resonances/kstar892analysis.cxx:744:100: error: unused parameter 'resotracks' [-Werror=unused-parameter]
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/6978/11ed463bd5be8f0496ad29fda72615e591878020/build_O2Physics_o2/pretty.html).

murioni commented:
> Hello @murioni , Can we consider adding an option of trackOccupancyInTimeRange cut on the initializer or CollisionCuts? One of the main idea of using the initializer is to handle all event selections on the initializer and minimize the analysis task. If the trackOccupancyInTimeRange value is needed to be computed in the analysis task, we can save it on the collision table.

Hello @BongHwi ,
I've added the occupancy cut to the CollisionCuts.h and added a configurable in ResonanceInitializer to apply this cut.
Thank you
